templates/identity/email-password/register.html.ejs

```diff
@@ -12,58 +12,184 @@
+                <label for="podName">Pod name:</label>
                 <input id="podName" type="text" name="podName" <% if (prefilled.podName) { %> value="<%= prefilled.podName %>" <% } %> />
               </li>
             </ol>
           </li>
         </ol>
       </fieldset>
 
+      <fieldset>
+        <legend>Your account</legend>
+        <div>
+          <p>
+            Choose the credentials you want to use to log in to this server in the future.
+          </p>
+          <ol>
+            <li>
+              <label for="email">Email:</label>
+              <input id="email" type="text" name="email" <% if (prefilled.email) { %> value="<%= prefilled.email %>" <% } %> />
+            </li>
+          </ol>
+          <ol id="passwordForm">
+            <li>
+              <label for="password">Password:</label>
+              <input id="password" type="password" name="password" />
+            </li>
+            <li>
+              <label for="confirmPassword">Confirm password:</label>
+              <input id="confirmPassword" type="password" name="confirmPassword" />
+            </li>
+          </ol>
+        </div>
+        <div id="noPasswordForm" class="hidden">
+          <p>
+            Since you will be using your existing WebID setup to access your pod,
+            <br>
+            you do <em>not</em> need to set a password.
+          </p>
+        </div>
+      </fieldset>
+
       <p class="actions"><button type="submit" name="submit">Sign up</button></p>
     </form>
+
+    <script>
+    // Assist the user with filling out the form by hiding irrelevant fields
+    (() => {
+      // Wire up the UI elements
+      const elements = {};
+      [
+        'createWebIdOn', 'createWebIdOff', 'createWebIdForm', 'existingWebIdForm', 'webId',
+        'createPod', 'createPodForm', 'podName',
+        'register', 'passwordForm', 'noPasswordForm', 'mainForm',
+      ].forEach(id => {
+        elements[id] = document.getElementById(id);
+        elements[id].addEventListener('change', updateUI);
+      });
+      updateUI();
+      mainForm.classList.add('loaded');
+
+      // Updates the UI when something has changed
+      function updateUI({ srcElement } = {}) {
+        // When Pod creation is required, automatically tick the corresponding checkbox
+        if (elements.createWebIdOn.checked)
+          elements.createPod.checked = true;
+        elements.createPod.disabled = elements.createWebIdOn.checked;
+
+        // Hide irrelevant fields
+        setVisibility('createWebIdForm', elements.createWebIdOn.checked);
+        setVisibility('existingWebIdForm', elements.createWebIdOff.checked);
+        setVisibility('createPodForm', elements.createPod.checked);
+        setVisibility('passwordForm', elements.createWebIdOn.checked ||
+          elements.createPod.checked || elements.register.checked);
+        setVisibility('noPasswordForm', !isVisible('passwordForm'));
```

Comment:
This means the `noPasswordForm` will only appear in case none of the 3 options is checked, which is an invalid request so this would never be relevant. To match the current server implementation `passwordForm` should only be visible if `elements.register.checked` is true. If we want to give the impression that we also use the password for pod management (which we might do in the future) then the password field should always be visible and the `noPasswordForm` field should instead contain information saying this is an invalid request.

Reply:
"none of the 3 options is checked" is not a possibility in the UI, because the first two options are connected radio buttons, forcing you to take one of the other.

Also note that the UI should encompass partial state. For example, wanting a pod but not having a name is invalid state. However, in order to get to the valid state "checked + name filled out", the UI has to temporarily be in an incomplete state since those two pieces of state need to be input sequentially.

Reply:
With none of the 3 options I mean: Using your existing WebID, unchecking the box to log in with this WebID and unchecking the box to create a new pod. This is a request that would do nothing on the server (and would error). This is the only state in which the `noPasswordForm` field appears.

What I'm saying is if we want to show the `noPasswordForm`, it should (also?) be when the `register` checkbox is checked. If we do want to show something in the above state it would be a statement saying that the request would error since nothing is being requested.

Reply:
It's shown in this state:

<img width="639" alt="image" src="https://user-images.githubusercontent.com/675313/127630591-29e2fe4d-7b20-4a64-92cc-67a9a3c8d3bd.png">

Is that correct?

Reply:
Probably not 😂

Reply:
> 
> 
> Probably not 😂

Yea, this sets all 3 booleans to false

Reply:
The new situation is now

```javascript
        setVisibility('passwordForm', elements.createWebIdOn.checked || elements.register.checked);
        setVisibility('noPasswordForm', !isVisible('passwordForm'));
```

which results in 

<img width="659" alt="image" src="https://user-images.githubusercontent.com/675313/127636331-22715ab3-2cda-48a8-8292-1c0384279da3.png">

I know the state shown in the screenshot is invalid, but this is an in-progress state. The user has to check one of the two boxes, but the UI cannot predict which of the two it will be.

If the user fails to make a choice, the server will get back with an error.

Reply:
(Note that `register` has a new meaning with the radio-button-based design.)

Reply:
> I know the state shown in the screenshot is invalid, but this is an in-progress state. The user has to check one of the two boxes, but the UI cannot predict which of the two it will be.
> 
> If the user fails to make a choice, the server will get back with an error.

Yea it doesn't have to say anything, I'm ok with that. It wouldn't have to predict though, it could just say a message like "Please pick at least one of the options" since everything is so dynamic anyway. But not really required.